src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+            } else if (userInputLowerCase.startsWith("event")) {
+                String taskToAdd = removeFirstWord(userInput);
+                addTask(TaskType.EVENT, taskToAdd, list);
+            } else {
+                System.out.println(HORIZONTAL_LINE + HELP_MESSAGE + HORIZONTAL_LINE);
+            }
+        }
+    }
+
+    /**
+     * Adds a task to an array list of tasks, printing out the newly added to task to the terminal
+     *
+     * @param type        The type of task (using the TaskType enumerator)
+     * @param description Description of the task to be added
+     * @param taskList    Array list of tasks
+     */
+    private static void addTask(TaskType type, String description, List<Task> taskList) {
+        switch (type) {
+        case TODO:
+            taskList.add(new Todo(description));
+            break;
+        case DEADLINE:
+            if (description.contains("/by")) {
+                String[] separated = splitDescriptionFromTiming(TaskType.DEADLINE, description);
+                taskList.add(new Deadline(separated[0], separated[1]));
+            } else {
+                System.out.println("You need to specify a deadline! TIP: Use \"/by\" to do so!\n" +
+                        HORIZONTAL_LINE);
+                return;
+            }
+            break;
+        case EVENT:
+            if (description.contains("/at")) {
+                String[] separated = splitDescriptionFromTiming(TaskType.EVENT, description);
+                taskList.add(new Event(separated[0], separated[1]));
+            } else {
+                System.out.println("You need to specify an event! TIP: Use \"/at\" to do so!\n" +
+                        HORIZONTAL_LINE);
+                return;
+            }
+            break;
+        default:
+            taskList.add(new Task(description));
+        }
+
+        Task newlyAddedTask = taskList.get(taskList.size() - 1);
+        System.out.println(HORIZONTAL_LINE + "Got it! I've added this task:\n" +
+                newlyAddedTask + "\n" + HORIZONTAL_LINE);
+    }
+
+    /**
+     * Split the description of a Task from its timing (e.g. deadline) if it has one
+     *
+     * @param type        The type of task
+     * @param description Full string input of the task and its timing
+     * @return Returns a string array with index 0 containing the task description and index 1 containing the timing
+     */
+    public static String[] splitDescriptionFromTiming(TaskType type, String description) {
+        String[] separated;
+        switch (type) {
+        case DEADLINE:
+            separated = description.split("/by +");
+            break;
+        case EVENT:
+            separated = description.split("/at +");
+            break;
+        default:
+            throw new IllegalStateException("Unexpected value: " + type.toString());
+        }
+        return separated;
+    }
+
+    public static void printList(List<Task> list) {
```

Comment:
maybe helper functions for printing can be refactored into new class DisplayManager